model-optimizer/extensions/ops/einsum.py

```diff
@@ -70,7 +86,12 @@ def parse_equation(node_name: str, equation: str) -> (list, str):
                     "The output subscript of Einsum node {} must contain ellipsis".format(node_name)
         elif len(splitted_equation) == 1:
             # recover output subscript in case implicit mode
-            output_subscript = ''.join(input_subscripts_list)
+            output_subscript = ""
+            for ind, input_subscript in enumerate(input_subscripts_list):
+                labels = Einsum.extract_subscript_labels(node_name, input_subscript)
+                for label in labels:
+                    if Einsum.is_label_elsewhere(input_subscripts_list, label, [ind]) is False:
```

Comment:
In your case `if` works if `Einsum.is_label_elsewhere(input_subscripts_list, label, [ind])` returns None. But I strictly require boolean `False` value and I am sure about boolean type because I have specified a function signature